examples/pkg/kcp/clients/informers/example/interface.go

```diff
@@ -0,0 +1,71 @@
+//go:build !ignore_autogenerated
+// +build !ignore_autogenerated
+
+/*
+Copyright The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Code generated by kcp code-generator. DO NOT EDIT.
+
+package example
+
+import (
+	"acme.corp/pkg/kcp/clients/informers/example/v1"
+	"acme.corp/pkg/kcp/clients/informers/example/v1alpha1"
+	"acme.corp/pkg/kcp/clients/informers/example/v1beta1"
+	"acme.corp/pkg/kcp/clients/informers/example/v2"
+	"acme.corp/pkg/kcp/clients/informers/internalinterfaces"
+)
+
+type ClusterInterface interface {
+	// V1 provides access to the shared informers in V1.
+	V1() v1.ClusterInterface
+	// V1alpha1 provides access to the shared informers in V1alpha1.
+	V1alpha1() v1alpha1.ClusterInterface
+	// V1beta1 provides access to the shared informers in V1beta1.
+	V1beta1() v1beta1.ClusterInterface
+	// V2 provides access to the shared informers in V2.
+	V2() v2.ClusterInterface
+}
+
+type group struct {
+	factory          internalinterfaces.SharedInformerFactory
+	tweakListOptions internalinterfaces.TweakListOptionsFunc
+}
+
+// New returns a new ClusterInterface.
+func New(f internalinterfaces.SharedInformerFactory, tweakListOptions internalinterfaces.TweakListOptionsFunc) ClusterInterface {
```

Comment:
maybe todo: have `f` as the only required param, move everything else to functional options